docs/Contributing/API-for-contributors.md

```diff
@@ -2394,6 +2404,61 @@ In regards to the `notifications` key:
 - `enforce_bitlocker_encryption` applies only to Windows devices and means that it should encrypt the disk and report the encryption key back to Fleet.
 
 
+#### Get device's software
+
+Lists the software installed on the current device.
+
+`GET /api/v1/fleet/device/{token}/software/titles`
```

Comment:
@marko-lisica just to make sure, this is NOT the same data as `Get host's software` below, right (`GET /api/v1/fleet/hosts/:id/software`)? We deliberately want different responses for those 2 endpoints? Asking because often times we have the same endpoint available as user-authenticated and token-authenticated, but here this is not the case.

Also, is this endpoint and `GET /api/v1/fleet/hosts/:id/software` premium-only? I don't think so as it's not mentioned in those docs, but this could be an oversight (I know the [story itself](https://github.com/fleetdm/fleet/issues/14921) is Premium-only, but I'm guessing those 2 endpoints are not).

Reply:
On closer look, the figma looks like the my device page shows a page very similar to the host's software page, shouldn't this endpoint return the same response payload as Get host's sofware? Also, we see in the figma that the results are paginated, and "Get host's software" is a paginated endpoint. I'm starting to think this should be the same endpoint as "Get host's software", but token-authenticated instead of user-authenticated (and maybe we'd want to call it `GET /api/v1/fleet/device/{token}/software` instead of `GET /api/v1/fleet/device/{token}/software/titles`)?

Reply:
> We deliberately want different responses for those 2 endpoints?
Hey @mna, the response for those 2 endpoints is different purposefully.`GET /api/v1/fleet/hosts/:id/software` returns data about the package and `last_install` which is necessary to show details about last installation in the UI. We omited those in `GET /api/v1/fleet/device/{token}/software/titles` since it's not used. We'll add additional data within [self-service story](https://github.com/fleetdm/fleet/issues/17587), that's in current design sprint.

These endpoints should be available to free users. For free users status will be `null` since they can't add and install software. Thanks for catching this.

Reply:
> maybe we'd want to call it GET /api/v1/fleet/device/{token}/software instead of GET /api/v1/fleet/device/{token}/software/titles

@mna I agree. We changed this for host's software but skipped this one. I'll change this now to: `GET /api/v1/fleet/device/{token}/software`